daffodil-lib/src/main/scala/passera/unsigned/ULong.scala

```diff
@@ -240,6 +241,19 @@ object ULong {
   val Zero = MinValue
   val MaxValue = ULong(~0L)
 
-  val MaxValueAsBigInt = JBigInt.valueOf(Long.MinValue).abs
-  private val maxULongString = MaxValueAsBigInt.toString()
+  val MaxValueAsBigInt = new JBigInt("FFFFFFFFFFFFFFFF", 16)
+
+  def apply(bi: JBigInt): ULong = {
+    val posBigInt = bi.and(ULong.MaxValueAsBigInt)
+    ULong(posBigInt.longValue)
+  }
+
+  def apply(digits: String, radix: Int): ULong = {
+    if (digits.contains("-") || digits.contains(","))
+      throw new IllegalArgumentException("digits must contain only 0-9A-Fa-F.")
```

Comment:
Can we rely on JBigInt to verify the correctness of the digits string + radix? And if the minus sign is a problem, we can error if bi is negative, but maybe there's even a reason to allow that?

Reply:
Good point. We should not check things here, and just let JBigInt do it. 
This library has that "style" where it tries to be overly accepting of arguments and "make them work somehow" rather than my preferred style, which is to be as narrow as possible.